echidna/core/dsextract.py

```diff
@@ -202,19 +219,25 @@ class EnergyExtractReco(Extractor):
       fv_radius (float, optional): Fiducial radius. Applies a cut to remove
         events which have a radial position greater than the radius of the
         fiducial volume. If None no cut is applied.
-      mc_pos (bool, optional): If true then  MC position is used for cuts.
+      external (bool, optional): Set to True for external backgrounds.
+        Default is False.
+      mc_pos (bool, optional): If True then  MC position is used for cuts.
         If False (default) then position cuts will be made on reconstructed
         position.
 
     Attributes:
-      _mc_pos (bool): If true then MC position is used for cuts.
+      _mc_pos (bool): If True then MC position is used for cuts.
         If False then position cuts will be made on reconstructed position.
     '''
 
-    def __init__(self, fv_radius=None, mc_pos=False):
+    def __init__(self, fv_radius=None, external=False, mc_pos=False):
         '''Initialise the class
         '''
-        super(EnergyExtractReco, self).__init__("energy_reco", fv_radius)
+        super(EnergyExtractReco, self).__init__("energy_reco", fv_radius,
+                                                external)
+        if external and mc_pos:
+            raise ValueError("External and mc_pos are True. Can't have mc_pos "
+                             "for externals.")
```

Comment:
Ah, OK yes this makes perfect sense now. You could perhaps add a short `note` to the Class documentation to explain that having both `external` and `mc_pos` as `True` is incompatible because it will cut all events.